sdk/python/feast/types.py

```diff
@@ -110,8 +117,35 @@ def __init__(self, base_type: Union[PrimitiveFeastType, ComplexFeastType]):
 
         self.base_type = base_type
 
-    def to_int(self) -> int:
+    def to_value_type(self) -> int:
         assert isinstance(self.base_type, PrimitiveFeastType)
         value_type_name = PRIMITIVE_FEAST_TYPES_TO_VALUE_TYPES[self.base_type.name]
         value_type_list_name = value_type_name + "_LIST"
         return ValueTypeProto.Enum.Value(value_type_list_name)
+
+
+def from_value_type(
+    value_type: ValueTypeProto.Enum,
+) -> Union[ComplexFeastType, PrimitiveFeastType]:
+    """
+    Converts a ValueTypeProto.Enum to a Feast type.
+
+    Args:
+        value_type: The ValueTypeProto.Enum to be converted.
+
+    Raises:
+        ValueError: The conversion could not be performed.
+    """
+    # Primitive types can be directly converted.
+    primitive_values = [primitive_type.value for primitive_type in PrimitiveFeastType]
+    if value_type in primitive_values:
+        return PrimitiveFeastType(value_type)
+
+    # Complex types must be constructed. Currently only arrays are supported. Note that
+    # enum values for arrays are precisely the enum value for the array's base type plus 10.
```

Comment:
Where does this 10 come from?

Reply:
Honestly a map is better here because this is sure to break if we add enums that aren't supported or something. We should have a map, and a test that iterates over all possible enum values and ensures that you get a valid feast type.